src/main.c

```diff
@@ -182,8 +182,13 @@ main (int    argc,
   ret = 0;
 
  out:
-  if (sigint_id > 0)
-    g_source_remove (sigint_id);
+  if (sigint_source)
+    {
+      if( !g_source_is_destroyed(sigint_source))
```

Comment:
I spent quite a bit of time trying to figure out why `g_source_remove (sigint_id)` isn't working, but I'm thinking now I should look into it some more.